persistence-api/src/main/java/io/stargate/db/cdc/CDCServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.cdc;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import com.codahale.metrics.MetricRegistry;
+import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
+import com.datastax.oss.driver.shaded.guava.common.util.concurrent.ThreadFactoryBuilder;
+import io.stargate.db.metrics.CDCMetrics;
+import java.util.concurrent.*;
+import org.apache.cassandra.stargate.db.MutationEvent;
+import org.apache.cassandra.stargate.exceptions.CDCWriteException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CDCServiceImpl implements CDCService {
+
+  private static final CompletableFuture<Void> completedFuture =
+      CompletableFuture.completedFuture(null);
+  private static final CompletableFuture<Void> unhealthyFuture = new CompletableFuture<>();
+  private static final Logger logger = LoggerFactory.getLogger(CDCService.class);
+
+  private final CDCProducer producer;
+  private final CDCHealthChecker healthChecker;
+  private final CDCConfig config;
+  private final ScheduledExecutorService timeoutScheduler =
```

Comment:
`1` thread for this tread pool maybe not enough. There will be high traffic on `publish` but `orTimeout` will have only one thread. The default `CompletableFuture` uses a fork-join pool that has more threads so maybe we should use a similar number here.

Reply:
My understanding is that openjdk's `orTimeout()` uses 1 thread: https://github.com/openjdk/jdk/blob/928da494a853c164452c96d01318c94d115ca946/src/java.base/share/classes/java/util/concurrent/CompletableFuture.java#L2825-L2829

We usually shouldn't care about the timeout accuracy and as long as the caller is async all the way, the invoker thread should do little effort.